hls-graph/src/Development/IDE/Graph/Internal/Types.hs

```diff
@@ -144,17 +144,17 @@ data Result = Result {
     resultData      :: !BS.ByteString
     }
 
-data ResultDeps = UnknownDeps | AlwaysRerunDeps !KeySet | ResultDeps !KeySet
+data ResultDeps = UnknownDeps | AlwaysRerunDeps ![KeySet] | ResultDeps ![KeySet]
```

Comment:
make it clear that these are stored in reverse order

Reply:
Ok, I added a comment about this invariant.
and it also occurred to me that `AlwaysRerunDeps` should recovered to `KeySet`, since it would be one way down to `AlwaysRerunDeps` on `<>`, and only `ResultDeps` need to maintain this invariant.